source/write-operations/bulk-writes.txt

```diff
@@ -1,58 +1,347 @@
+  operations, see the :manual:`Field Update Operators
+  </reference/operator/update-field/>` guide in the {+mdb-server+} manual.
+
+``UpdateOneModel`` updates *the first* document that matches your query filter.
+
+The following example creates an instance of ``UpdateOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.java
+   :start-after: start-bulk-update-one
+   :end-before: end-bulk-update-one
+   :language: java
+   :copyable:
+   :dedent:
+
+To update multiple documents, create an instance of ``UpdateManyModel`` and pass in
+the same arguments. ``UpdateManyModel`` updates *all* documents that match your query
+filter.
+
+The following example creates an instance of ``UpdateManyModel``:
+
+.. literalinclude:: /includes/write/bulk-write.java
+   :start-after: start-bulk-update-many
+   :end-before: end-bulk-update-many
+   :language: java
+   :copyable:
+   :dedent:
+
+Replace Operations
+~~~~~~~~~~~~~~~~~~
+
+A replace operation removes all fields and values of a specified document and
+replaces them with new ones. To perform a replace operation, create an instance
+of ``ReplaceOneModel`` and pass in a query filter and the fields and values you want
+to store in the matching document.
+
+The following example creates an instance of ``ReplaceOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.java
+   :start-after: start-bulk-replace-one
+   :end-before: end-bulk-replace-one
+   :language: java
+   :copyable:
+   :dedent:
+
+To replace multiple documents, you must create an instance of ``ReplaceOneModel`` for each document.
+
+Delete Operations
+~~~~~~~~~~~~~~~~~
+
+To delete a document, create an instance of ``DeleteOneModel`` and pass in a
+query filter specifying the document you want to delete. ``DeleteOneModel`` removes
+only *the first* document that matches your query filter.
+
+The following example creates an instance of ``DeleteOneModel``:
+
+.. literalinclude:: /includes/write/bulk-write.java
+   :start-after: start-bulk-delete-one
+   :end-before: end-bulk-delete-one
+   :language: java
+   :copyable:
+   :dedent:
+
+To delete multiple documents, create an instance of ``DeleteManyModel`` and pass in a
+query filter specifying the document you want to delete. ``DeleteManyModel`` removes
+*all* documents that match your query filter.
+
+The following example creates an instance of ``DeleteManyModel``:
+
+.. literalinclude:: /includes/write/bulk-write.java
+   :start-after: start-bulk-delete-many
+   :end-before: end-bulk-delete-many
+   :language: java
+   :copyable:
+   :dedent:
+
+Call the ``bulkWrite()`` Method
+--------------------------------
+
+After you define a class instance for each operation you want to perform,
+pass a list of these instances to the ``bulkWrite()`` method.
+By default, the method runs the operations in the order
+they're defined in the list.
+
+The following example performs multiple write operations by using the
+``bulkWrite()`` method:
+
+.. io-code-block::
```

Comment:
I: looks like there's an rST error that's preventing the io code block from rendering. I'm not sure what the issue is; maybe remove the :copyable: and see if that works?